lib/formtastic/helpers/inputs_helper.rb

```diff
@@ -350,12 +350,21 @@ def association_columns(*by_associations) # @private
         end
       end
 
+      # Collects non-standard foreign key columns
+      def foreign_key_columns
+        if @object.present? && @object.class.respond_to?(:reflections)
+          @object.class.reflections.map {|_,v| v.options[:foreign_key] }.compact
```

Comment:
Or include the reflection methods to those models. They are in `ActiveRecord::Reflection::ClassMethods`.

What I meant was create instance like `ActiveRecord::Reflection::BelongsToReflection.new` instead of creating mocks. That way we can really guarantee rails compatibility.